src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -228,39 +227,12 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
     Context previous = withMetadata.attach();
     Profiler prof = Profiler.instance();
     try {
-      try {
-        // Try to lookup the action in the action cache.
-        ActionResult cachedResult;
-        try (SilentCloseable c = prof.profile(ProfilerTask.REMOTE_CACHE_CHECK, "check cache hit")) {
-          cachedResult = acceptCachedResult ? remoteCache.downloadActionResult(actionKey) : null;
-        }
-        if (cachedResult != null) {
-          if (cachedResult.getExitCode() != 0) {
-            // Failed actions are treated as a cache miss mostly in order to avoid caching flaky
-            // actions (tests).
-            // Set acceptCachedResult to false in order to force the action re-execution
-            acceptCachedResult = false;
-          } else {
-            try {
-              return downloadAndFinalizeSpawnResult(
-                  cachedResult, /* cacheHit= */ true, spawn, context, remoteOutputsMode);
-            } catch (CacheNotFoundException e) {
-              // No cache hit, so we fall through to local or remote execution.
-              // We set acceptCachedResult to false in order to force the action re-execution.
-              acceptCachedResult = false;
-            }
-          }
-        }
-      } catch (IOException e) {
-        return execLocallyAndUploadOrFail(
-            spawn, context, inputMap, actionKey, action, command, uploadLocalResults, e);
-      }
-
       ExecuteRequest.Builder requestBuilder =
           ExecuteRequest.newBuilder()
               .setInstanceName(remoteOptions.remoteInstanceName)
               .setActionDigest(actionKey.getDigest())
-              .setSkipCacheLookup(!acceptCachedResult);
+              // Always skip cache lookup as this is done in AbstractSpawnStrategy
+              .setSkipCacheLookup(true);
```

Comment:
I wouldn't make this change. Note that the old code also made two lookups, so skipping here is a significant departure from before.

However, I also think that Bazel shouldn't call the remote cache at all when remote execution is enabled and instead directly try to execute the action. The cache lookup is an unnecessary additional round-trip.